Highbar.Types.Test/ValidationTests.cs

```diff
@@ -0,0 +1,1284 @@
+          Assert.Equal(expectedValidation.ToString(), testValidation.ToString());
+        }
+
+        [Fact]
+        public void ShouldReturnSuccessForJust()
+        {
+          Maybe<bool> testMaybe = Maybe<bool>.Just(_testValue);
+          Validation<Exception, bool> testValidation = Validation<Exception, bool>.From(testMaybe);
+          Validation<Exception, bool> expectedValidation = Validation<Exception, bool>.Pass<Exception, bool>(_testValue);
+
+          Assert.Equal(expectedValidation, testValidation);
+        }
+
+        [Fact]
+        public void ShouldReturnFailureForNullValidation()
+        {
+          Validation<Exception, bool> testValue = null;
+          Validation<Exception, bool> testValidation = Validation<Exception, bool>.From(testValue);
+          Validation<Exception, bool> expectedValidation = Validation<Exception, bool>.Fail<Exception, bool>(new NullReferenceException("value is null"));
+
+          //NOTE(lee.crabtree):  The underlying exceptions only test for reference equality.  (/facepalm)
+          Assert.Equal(expectedValidation.ToString(), testValidation.ToString());
+        }
+
+        [Fact]
+        public void ShouldReturnFailureForFailure()
+        {
+          Validation<string, bool> testValidation = Validation<string, bool>.Fail<string, bool>("Fail");
+          Validation<string, bool> expectedValidation = Validation<string, bool>.Fail<string, bool>("Fail");
+          Validation<string, bool> actualValidation = Validation<string, bool>.From(testValidation);
+
+          Assert.Equal(expectedValidation, actualValidation);
+        }
+
+        [Fact]
+        public void ShouldReturnSuccessForSuccess()
+        {
+          Validation<string, bool> testValidation = Validation<string, bool>.Pass<string, bool>(_testValue);
+          Validation<string, bool> expectedValidation = Validation<string, bool>.Pass<string, bool>(_testValue);
+          Validation<string, bool> actualValidation = Validation<string, bool>.From(testValidation);
+
+          Assert.Equal(expectedValidation, actualValidation);
+        }
+      }
+
+      public class FromFailure
+      {
+        [Fact]
+        public void ShouldReturnValueForFailure()
+        {
+          Validation<string, bool> testValidation = Validation<string, bool>.Fail<string, bool>(_testError);
+          IEnumerable<string> expectedErrors = new List<string> { _testError };
+          IEnumerable<string> actualErrors = Validation<string, bool>.FromFailure(testValidation);
+
+          Assert.Equal(expectedErrors, actualErrors);
+        }
+
+        [Fact]
+        public void ShouldThrowArgumentExceptionForSuccess()
+        {
+          Validation<string, bool> testValidation = Validation<string, bool>.Pass<string, bool>(_testValue);
+          Exception exception = Record.Exception(() => Validation<string, bool>.FromFailure(testValidation));
+
+          Assert.NotNull(exception);
+          Assert.IsType<ArgumentException>(exception);
```

Comment:
Sure would be nice if this would also include the `null` check to avoid multiple asserts.